benchmarks/adler32.bril

```diff
@@ -0,0 +1,84 @@
+@main {
+  size: int = const 512;
+  arr: ptr<int> = alloc size;
+  call @fill_array arr size;
+  checksum: int = call @adler32 arr size;
+  print checksum;
+  free arr;
+}
+
+# Calculate the modulo of the two numbers using arithmetic
+@mod(r: int, s: int): int {
+  x: int = div r s;
+  y: int = mul x s;
+  result: int = sub r y;
+  ret result;
+}
+
+# Fills the array with values 0, 1. 2, ...
+@fill_array(arr: ptr<int>, size:int) {
+  one: int = const 1;
+  curr: int = const 0;
+  loc: ptr<int> = id arr;
+.loop:
+  store loc curr;
+  loc: prt<int> = ptradd loc one;
```

Comment:
From brilirs: `prt` -> `ptr`

Reply:
Interesting that `brili` didn't catch this.

Reply:
My understanding is that `brili` does it's type checking based on the type of the value being evaluated and ignores the destination type.

Reply:
Very true; brili is purely dynamic. There are static type checkers elsewhere in the repository.